Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -48,15 +48,8 @@ class CustomerIOImplementation: CustomerIOInstance {
     }
 
     public var profileAttributes: [String: Any] {
-        get {
-            [:]
-        }
-        set {
-            guard let existingProfileIdentifier = profileStore.identifier else {
-                return
-            }
-            identify(identifier: existingProfileIdentifier, body: newValue)
-        }
+        get { DataPipeline.shared.profileAttributes }
+        set { DataPipeline.shared.profileAttributes = newValue }
```

Comment:
This is OK for now, but I want to point out to possibly avoid more tech debt in future PRs. 

By the Tracking module using a singleton instance directly inside of function bodies like this will make this function body untestable. 

I faced a similar situation recently when the messagingpush module needed to call `CustomerIO.shared` in the track module. [I used `CustomerIOInstance` protocol](https://github.com/customerio/customerio-ios/blob/b61b6d52b87f02ccb5f406bd559f7ff17e0d3411/Sources/MessagingPush/PushHandling/PushClickHandler.swift#L17-L19) in the class so that I could mock it in automated tests. 

[I then added a DiGraph extension](https://github.com/customerio/customerio-ios/blob/fb7b75d93563e9ab97692a31d4365131b161306b/Sources/Tracking/CustomerIO.swift#L427-L436) so that `DataPipeline.shared` could be used in production when not mocking. 

This is a non-blocker. Sharing idea if it's helpful for future PRs.